include/aspect/material_model/equation_of_state/multicomponent_incompressible.h

```diff
@@ -0,0 +1,118 @@
+       * length N+1, where N is the number of compositional fields used in the computation.
+       * The first entry corresponds to the "background" (which is also why there are N+1 entries).
+       *
+       * If a single value is given, then all the compositional fields are given
+       * that value. Other lengths of lists are not allowed. For a given
+       * compositional field the material parameters are treated as constant,
+       * except density, which varies linearly with temperature according to the equation:
+       *
+       * $\rho(p,T,\mathfrak c) = \left(1-\alpha_i (T-T_0)\right) \rho_0(\mathfrak c_i).$
+       *
+       * There is no pressure-dependence of the density.
+       */
+      template <int dim>
+      class MulticomponentIncompressible :  public ::aspect::SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * A function that computes the output of the equation of state @p out
+           * for all compositions, given the inputs in @p in and an index q that
+           * determines which entry of the vector of inputs is used.
+           */
+          void evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+                        const unsigned int q,
+                        MaterialModel::EquationOfStateOutputs<dim> &out) const;
+
+          /**
+           * Return whether the model is compressible or not. Incompressibility
+           * does not necessarily imply that the density is constant; rather, it
+           * may still depend on temperature or pressure. In the current
+           * context, compressibility means whether we should solve the continuity
+           * equation as $\nabla \cdot (\rho \mathbf u)=0$ (compressible Stokes)
+           * or as $\nabla \cdot \mathbf{u}=0$ (incompressible Stokes).
+           */
+          bool is_compressible () const;
+
+          /**
+           * Declare the parameters this class takes through input files.
+           * The optional parameter @p n_compositions determines the maximum
+           * number of compositions the equation of state is set up with,
+           * in other words, how many compositional fields influence the
+           * density.
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm,
+                              const double default_thermal_expansion = 3.5e-5);
+
+          /**
+           * Read the parameters this class declares from the parameter file.
+           * The optional parameter @p n_compositions determines the maximum
+           * number of compositions the equation of state is set up with,
+           * and should have the same value as the parameter with the same
+           * name in the declare_parameters() function.
+           */
+          void
+          parse_parameters (ParameterHandler &prm);
+
+
+        private:
+          // Vector of reference densities $\rho_0$ with one entry per composition,
```

Comment:
Make these comments into doxygen documentation: /** ... */. Same for all parameters below.